### PR TITLE
python3Packages.web3: update to 6.3.0

### PR DESCRIPTION
**Pull request by hellwolf**
###### Description of changes

Update to web3.py 6.3.0.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hellwolf** [comment]
@ofborg build python311Packages.web3 python310Packages.web3

File: pkgs/development/python-modules/web3/default.nix [hunk 1]
```diff
@@ -72,6 +72,10 @@ buildPythonPackage rec {
 
   doCheck = false;
 
+  postPatch = ''
+    substituteInPlace setup.py --replace "types-protobuf==3.19.13" "types-protobuf"
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/web3/default.nix:76]: I think you want `pythonRelaxDeps`

**Review by RaitoBezarius — APPROVED** [review]

File: pkgs/development/python-modules/web3/default.nix (hunk 1 above)
Comment (hellwolf) [reply pkgs/development/python-modules/web3/default.nix:76]: thanks! let me try

Comment (hellwolf) [reply pkgs/development/python-modules/web3/default.nix:76]: oh but that `pythonRelaxDeps` seem not affecting the `extras_require .linter` in https://github.com/ethereum/web3.py/blob/main/setup.py

I  am not familiar with nix python packaging yet.

Comment (RaitoBezarius) [reply pkgs/development/python-modules/web3/default.nix:76]: You're right, it's fine like this, let's move on :)

**Comment by RaitoBezarius** [comment]
ugh, also @hellwolf please follow CONTRIBUTING guide next time. I merged this when I should not have.

`python3Packages.web3: 5.31.0 -> 6.3.0` for the commit messages.

**Comment by hellwolf** [comment]
> ugh, also @hellwolf please follow CONTRIBUTING guide next time. I merged this when I should not have.
> 
> `python3Packages.web3: 5.31.1 -> 6.3.0` for the commit messages.

Yea, my bad, I just noticed, sorry

**Pull request merged**